tutorials/hoon/graph.md

```diff
@@ -0,0 +1,153 @@
++++
+title = "Graph Store"
+weight = 3
+template = "doc.html"
++++
+
+# What is Graph Store?
+
+Graph store is a gall implementation of a tree-based database that has an affordance for cyclic structures by allowing nodes to reference other nodes. It is most comparable to modern NoSQL databases in structure, though it allows for schemas to be written in the form of marks. It is used as the backend for an increasing number of Landscape applications, and is considered to be best used for social data.
+
+## Data types
+
+`graphs` is a map of `resource` to `marked-graph`
+
+`resource` is a pair of ship and term which is where the graph is hosted and the name of the resource. Every graph is it's own resource.
+
+`marked-graph` is a pair of `graph` and `(unit mark)`. The semantics of a graph are defined by the mark. This is just for validating the structure of the graph. 
+
+`graph` is a `mop`, a mold builder for order map, of an atom and a `node`.
+
+`node` is a pair of `post` and `children`.
+
+`post` is defined in `sur/post`. It contains:
+
+`author=ship`
+`index`
+`time-sent=time`
+`contents=(list content)`
+`hash=(unit hash)`
+`signatures`
+
+Most of these elements are self explanatory.
+
+`contents` is a `list` of `content` where `content` is a tagged union of text, url, code, or reference. This is where the contents of a publish post or an entry from links would be stored.
+
+`hash` is a `sham` hash of `validated-portion` which includes the parent hash, author, time sent, and contents.
+
+`signatures` is a set of `signature` which is used to cryptographically sign graph-store messages.
+
+When interacting with `graph-store` you will need to create an `update`
+
+`update` is a tagged union that currently only has one option
+
+`[%0 p=time q=update-0]`
+
+The reason this is a union is to serve as future proofing for version changes.
+
+An `update-0` can either be logged in the case of adding and removing nodes or signatures or non logged as in the case of adding or removing a graph. This logging refers to if this action adds an entry to the update log that is used to synchronize state between ships.
```

Comment:
> An `update-0` can either be logged in the case of adding and removing nodes or signatures or non logged as in the case of adding or removing a graph. 

Some punctuation
"An `update-0` can either be logged, as in the case of adding/removing nodes or signatures, or non-logged, as in the case of adding/removing a graph. "